lib/Doctrine/DBAL/Schema/Column.php

```diff
@@ -118,9 +119,10 @@ public function setOptions(array $options)
     {
         foreach ($options as $name => $value) {
             $method = "set".$name;
-            if (method_exists($this, $method)) {
-                $this->$method($value);
+            if (!method_exists($this, $method)) {
+                throw InvalidArgumentException::fromUnsupportedOption($name);
```

Comment:
Might be considered a BC-break, tell me if you want me to trigger deprecation errors instead.

Reply:
I notice that it does not seem to be used anywhere in the code, but maybe you're not against it?

Reply:
Yes, this could indeed be considered unnecessary BC break in 2.x for no real benefit... And I'm not even sure deprecation notice is worth it. Maybe we should just retarget this to develop/3.0?

Reply:
How about retargetting only this part, and merging the rest in master, since it's BC?

Reply:
> for no real benefit...

A smooth upgrade path for people misusing this method is a good benefit IMHO. And the number of misuses I found in `doctrine/dbal` alone suggests this might be useful to many, don't you think.

Reply:
Agreed, I meant there is _"no real benefit"_ in throwing an exception in 2.x due possible BC concerns.
Anyway, I'm not sure what is the correct approach regarging deprecations. I think the silent variant (what i.e. Symfony does) `@triger_error('blahblah', E_USER_DEPRECATED)` would be just fine. We will eventually need to deprecate multiple things anyway at latest in last 2.x before 3.0 comes (especially in ORM).

Reply:
Great, I'll go with that then!

Reply:
It's funny to see old options in the tests, that had no use at all. I like the stricter handling with that exception, but I agree on BC and the deprecation trigger.

Reply:
To test it properly, I will need to make `simple-phpunit` work though :(
Right now I get:

```
PHP Fatal error:  Class 'Doctrine\Tests\DbalTestCase' not found in /home/greg/dev/dbal/tests/Doctrine/Tests/DBAL/Cache/QueryCacheProfileTest.php on line 9

Fatal error: Class 'Doctrine\Tests\DbalTestCase' not found in /home/greg/dev/dbal/tests/Doctrine/Tests/DBAL/Cache/QueryCacheProfileTest.php on line 9

```